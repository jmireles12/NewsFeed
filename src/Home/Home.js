import React, { Component } from 'react'
import './Home.css'
import { Link } from 'react-router-dom'
import ApiContext from '../ApiContext'

export default class Home extends Component {
    static defaultProps = {
        match: {
            params: {}
        }
    }
    
    static contextType = ApiContext

    render() {
        return (
            <header className='home'>
                <h2 className='welcome__title'>Welcome</h2>
                <p className='welcome__message'>Welcome to NewsFeed where you can add Tabs
                and display news containing the name of the Tabs.
                I created this website to make searching news a
                little bit easier. Instead of having different tabs
                in your browser where you could get different articles
                or blogs, NewsFeed will only show you news and you
                can go back and forward from the Tabs that you created
                from the drop down. In 'My News' just press the edit button and you'll be
                able to create a tab with the name of your choosing.</p>
                <Link to='/mynews'><button>Go to News</button></Link>
            </header>
        )
    }
}