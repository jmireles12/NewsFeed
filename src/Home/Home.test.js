import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Home from './Home'

describe(`Home component`, () => {
    it('renders the complete Home', () => {
        const wrapper = shallow(<Home />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})