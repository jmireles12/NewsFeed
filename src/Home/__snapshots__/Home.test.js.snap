// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders the complete Home 1`] = `
<header
  className="home"
>
  <h2
    className="welcome__title"
  >
    Welcome
  </h2>
  <p
    className="welcome__message"
  >
    Welcome to NewsFeed where you can add Tabs and display news containing the name of the Tabs. I created this website to make searching for news a little bit easier. Instead of having different tabs in your browser where you could get different articles or blogs, NewsFeed will only show you news and you can go back and forward from the Tabs that you created from the drop-down. In 'My News' just press the edit button and you'll be able to create a tab with the name of your choosing.
  </p>
  <Link
    to="/mynews"
  >
    <button>
      Go to News
    </button>
  </Link>
</header>
`;
